lib/tasks/jenkins.rake

```diff
@@ -1,25 +1,24 @@
 begin
   require "ci/reporter/rake/minitest"
 
-  namespace :foreman do
-    task :set_test_runner do
-      ENV['TESTOPTS'] = "#{ENV['TESTOPTS']} #{Rails.root}/test/test_runner.rb"
-    end
-  end
-
   namespace :jenkins do
-    task :unit => ["jenkins:setup:minitest", 'rake:test:units', 'rake:test:lib', 'rake:test:functionals']
-    task :integration => ["jenkins:setup:minitest", 'rake:test:integration']
-    task :lib => ["jenkins:setup:minitest", 'rake:test:lib']
-    task :functionals => ["jenkins:setup:minitest", 'rake:test:functionals']
-    task :units => ["jenkins:setup:minitest", 'rake:test:units']
+    task :unit => ['jenkins:db', 'jenkins:setup:minitest',
+                   'rake:test:units', 'rake:test:lib', 'rake:test:functionals']
+    task :integration => ['jenkins:db', 'jenkins:setup:minitest', 'rake:test:integration']
+    task :lib => ['jenkins:db', 'jenkins:setup:minitest', 'rake:test:lib']
+    task :functionals => ['jenkins:db', 'jenkins:setup:minitest', 'rake:test:functionals']
+    task :units => ['jenkins:db', 'jenkins:setup:minitest', 'rake:test:units']
 
     namespace :setup do
       task :pre_ci do
         ENV["CI_REPORTS"] = 'jenkins/reports/unit/'
         gem 'ci_reporter'
       end
-      task :minitest  => [:pre_ci, "ci:setup:minitest", "foreman:set_test_runner"]
+      task :minitest  => [:pre_ci, "ci:setup:minitest"]
+    end
+
+    task :db do
+      system('rake db:migrate RAILS_ENV=test')
```

Comment:
1. No error handling, it appears to exit with a zero exit code when this step fails.
2. This duplicates the number of migrations needed compared to using db:test:prepare (which generates it from schema.rb).  If my comment above about using db:test:prepare doesn't obsolete this, then it should perhaps load the schema.

The schema's slightly preferable as a couple of years ago we hit an obscure bug where Rails migrated differently in development vs test environments (it disabled some SQL queries in test).  We moved to the more standard way of migrating development and relying on test DB creation from the schema for that reason, never migrating in the test env.